src/kdl/CMakeLists.txt

```diff
@@ -27,6 +27,8 @@ get_property(IDYNTREE_TREE_INCLUDE_DIRS GLOBAL PROPERTY IDYNTREE_TREE_INCLUDE_DI
 target_include_directories(${libraryname} PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${IDYNTREE_TREE_INCLUDE_DIRS}>"
                                                  "$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}>")
 
+target_compile_options(${libraryname} PUBLIC "-Wno-deprecated")
```

Comment:
Can you wrap this in a logic to check if the compiler is GCC or Clang (see https://github.com/robotology/idyntree/blob/2470a71cee49ee3a611195e36b8dc447eca9f154/cmake/iDynTreeOptions.cmake#L100)? 
This will avoid that the compilation fails on MSVC.